creds/ssm/manager.go

```diff
@@ -0,0 +1,132 @@
+package ssm
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"text/template"
+	"text/template/parse"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/concourse/atc/creds"
+)
+
+const DefaultSecretTemplate = "concourse-{{.Team}}/{{.Pipeline}}/{{.Secret}}"
+const DefaultFallbackTemplate = "concourse-{{.Team}}/{{.Secret}}"
+
+type SsmManager struct {
+	AwsAccessKeyID     string `long:"aws-access-key" description:"AWS Access key ID"`
+	AwsSecretAccessKey string `long:"aws-secret-key" description:"AWS Secret Access Key"`
+	AwsSessionToken    string `long:"aws-session-token" description:"AWS Session Token"`
+	AwsRegion          string `long:"aws-region" description:"AWS region to send requests to. Enviroment variable AWS_REGION is used if this flag is not provided."`
+	SecretTemplate     string `long:"secret-template" description:"AWS SSM parameter name template" default:"concourse-{{.Team}}/{{.Pipeline}}/{{.Secret}}"`
```

Comment:
I thought that if you reference a `((secret-value))` in a pipeline.yml, the current implementation of this (and other) secret backends will only check two paths:

- `/concourse/{{.Team}}/{{.Pipeline}}/secret-value`
- `/concourse/{{.Team}}/secret-value`

And that it's not possible to put e.g. `((nested/secret-value))` in pipeline.yml, and then retrieve `/concourse/{{.Team}}/{{.Pipeline}}/nested/secret-value`?

If I'm not mistaken about the above, I wonder if the users would actually be able to make use of the levels we save them by using `concourse-` as a prefix? (I.e., users would only run into trouble if they wanted to nest the concourse secrets under an existing path, which would require passing a new template via the command line anyway?)

Reply:
Thanks for all the feedback! I shall focus on topgun next.